server/service/microsoft_mdm.go

```diff
@@ -1248,17 +1249,125 @@ func (svc *Service) isTrustedRequest(ctx context.Context, reqSyncML *fleet.SyncM
+	if err != nil {
+		return err
+	}
+
+	if !fleetdPresent {
+		// TODO: Command should be enqueued here once the new commands tables are in place
+		return nil
+	}
+
+	return nil
+}
+
+// Generic Alert Handlers
+// This handler will check for generic alerts. Device unenrollment is handled here
+func (svc *Service) processGenericAlert(ctx context.Context, messageID string, deviceID string, cmd mdm_types.ProtoCmdOperation) error {
+	// Checking user-initiated unenrollment request
+	if len(cmd.Cmd.Items) > 0 {
+		for _, item := range cmd.Cmd.Items {
+
+			if item.Meta == nil || item.Meta.Type == nil || item.Meta.Type.Content == nil {
+				continue
+			}
+
+			// Checking if user-initiated unenrollment request is present
+			if *item.Meta.Type.Content == mdm.AlertUserUnenrollmentRequest {
+
+				// Deleting the device from the list of enrolled device
+				err := svc.ds.MDMWindowsDeleteEnrolledDeviceWithDeviceID(ctx, deviceID)
+				if err != nil {
+					return fmt.Errorf("unenrolling windows device: %w", err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
 // processIncomingAlertsCommands will process the incoming Alerts commands.
-// These commands requires an status response.
-func (svc *Service) processIncomingAlertsCommands(messageID string, deviceID string, cmd mdm_types.ProtoCmdOperation) (*fleet.SyncMLCmd, error) {
-	// TODO: New Session Initiation request and Device unenrollment should happen here
-	return NewSyncMLCmdStatus(messageID, cmd.Cmd.CmdID, cmd.Verb, mdm.CmdStatusOK), nil
+// These commands don't require an status response.
+func (svc *Service) processIncomingAlertsCommands(ctx context.Context, messageID string, deviceID string, cmd mdm_types.ProtoCmdOperation) error {
+	if cmd.Cmd.Data == nil {
+		return errors.New("invalid alert command")
+	}
+
+	// gathering the incoming Alert ID
+	alertID := *cmd.Cmd.Data
+
+	switch alertID {
+	case mdm.CmdAlertClientInitiatedManagement:
+		return svc.processNewSessionAlert(ctx, messageID, deviceID, cmd)
+	case mdm.CmdAlertServerInitiatedManagement:
+		return svc.processNewSessionAlert(ctx, messageID, deviceID, cmd)
+	case mdm.CmdAlertGeneric:
+		return svc.processGenericAlert(ctx, messageID, deviceID, cmd)
+	}
+
+	return nil
 }
 
 // processIncomingResultsCommands will process the incoming Results commands.
 // These commands requires don't require an status response.
-func (svc *Service) processIncomingResultsCommands(deviceID string, cmd mdm_types.ProtoCmdOperation) (*fleet.SyncMLCmd, error) {
-	// TODO: Results of operations such as GET and EXEC should be processed here
+func (svc *Service) processIncomingResultsCommands(ctx context.Context, sessionID string, deviceID string, cmd mdm_types.ProtoCmdOperation) (*fleet.SyncMLCmd, error) {
+	if cmd.Cmd.MsgRef == nil || cmd.Cmd.Items == nil {
+		return nil, errors.New("invalid results command")
+	}
+
+	msgRef := *cmd.Cmd.MsgRef
+	msgData := ""
+	// Checking user-initiated unenrollment request
```

Comment:
Is this comment a copy-paste from elsewhere? Not sure if this is related to unenrollment?